components/omega/doc/design/OmegaV0ShallowWater.md

```diff
@@ -0,0 +1,700 @@
+The three kites sum to the triangle area,
+$$
+{\hat A}_v = \sum_{i\in CV(v)} {\tilde A}_{v,i}.
+$$
+
+#### 3.2.9. Vector from Edge to Cell
+
+The prognostic velocity variable on the edge is the edge normal velocity, $u_e$. The tangential velocity $u_e^\perp$ is computed diagnostically from $u_e$. In addition, the full vector may be computed at the cell center from the edge normal velocities $u_e$ of the edges on that cell. That is done in MPAS with radial basis functions, and is explained in this [previous design document](https://github.com/MPAS-Dev/MPAS-Documents/blob/master/shared/rbf_design/rbf.pdf).
+
+### 3.3 Momentum Terms
+
+The computation of each term in (4-6) is now described in detail, along with alternative formulations.
+
+#### 3.3.1. Kinetic energy gradient
+
+The kinetic energy gradient term is the non-rotational part of the nonlinear advection. Fundamentally, it maps the prognostic edge-normal velocity from edges back to an edge scalar quantity. We use the standard gradient formulation from cell center to edge,
+
+$$
+\nabla K_i = \frac{1}{d_e} \sum_{i\in CE(e)} -n_{e,i}K_i.
+$$
+
+There are several methods to compute the cell-centered kinetic energy. The first uses the edge-normal velocities as in
+[Ringler et al. 2010](https://www.sciencedirect.com/science/article/pii/S0021999109006780) equation 63,
+
+$$
+K_i = \frac{1}{A_i} \sum_{e\in EC(c)} \frac{{\bar A}_{e}}{4}u_e^2.
+$$
+
+Here ${\bar A}_{e}$ is the rectangle about the edge defined by ${\bar A}_{e} = l_e d_e$ so that
+$$
+A_i = \sum_{e\in EC(c)} \frac{{\bar A}_{e}}{4}
+$$
+as stated in 
+[Ringler et al. 2010](https://www.sciencedirect.com/science/article/pii/S0021999109006780) equations 48, 64, and footnote 4.
+
+One might expect an additional factor of one-half in the $K_i$ formula, as in the standard formula for kinetic energy, $K = \tfrac{1}{2} m v^2$. However, the one-half is missing because $u_e$ is only one component of the velocity vector on the edge. One could replace $u_e^2$ with $\tfrac{1}{2}(u_e^2 + {u_e^{\perp}}^2)$, but $u^\perp_e$ is already computed from the surrounding values of $u_e$, so we don't do that because it would just add redundant information. As a side note, the mass $m$ is missing because the mass is already divided out of the momentum equation (1).
+
+The kinetic energy may also be computed at the vertex.  This uses the velocity on the three connected edges,
+$$
+K_v = \frac{1}{A_v} \sum_{e\in EV(v)} \frac{{\bar A}_{e}}{4}u_e^2.
+$$
+The normalization can be found through simple geometric arguments to be similar to above,
+$$
+A_v = \sum_{e\in EV(v)} \frac{{\bar A}_{e}}{4}.
+$$
+
+The computation of $K_v$ can be found in the MPAS-Ocean code with the `kineticEnergyVertexOnCells` variable. The final kinetic energy at the cell center may be chosen to be either $K_i$ in the first equation above, or a linear combination of the two. In MPAS-Ocean, the flag `config_include_KE_vertex` uses the value
+
+$$
+\frac{5}{8} K_i + \frac{3}{8} [K_v]_i
```

Comment:
Vertex to cell interpolation (`[K_v]_i`) is not defined in the document.

Reply:
Added that. thanks.